src/qibolab/platforms/platform.py

```diff
@@ -85,3 +115,51 @@ def set_bias(self, qubit, bias):
 
     def get_bias(self, qubit):
         return self.qubits[qubit].flux.bias
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+
+    CYLIC: Better averaging for noise,
+    SINGLESHOT: False averaging,
+    [SEQUENTIAL: Worse averaging for noise]
+
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+
+
+@dataclass
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: Number of shots per point on the experiment
+    :relaxation_time: Relaxation time for the qubit
+    :fast_reset: Enable or disable fast reset
+    :acquisition_type: Data acquisition mode
+    :averaging_mode: Data averaging mode
+    """
+
+    nshots: Optional[np.uint32] = 1024
+    relaxation_time: Optional[np.uint32] = 100e-9
```

Comment:
This should be
```suggestion
    relaxation_time: Optional[np.uint32] = None
```
so that by default we use the `relaxation_time` provided by the platform/runcard.

Also it may be useful to enforce specific units (eg. in the docstring) for times, however this is not strictly necessary if you are consistent with units when using a specific platform. I think in most platforms we are using <ns> but it should be fine to use a different unit in your platform as long as you don't mix different units in the same platform. If we don't use same units for all platforms it is even more important to use `None` as default here.

Side note: you are using int type but 100e-9 is not int.

Reply:
Sure, I would leave as a float as even if we had the time in ns we may want to smaller than that. All the other platforms, do they use ns or s as their time unit ?